src/main/java/data_access/InMemoryUserDataAccessObject.java

```diff
@@ -42,6 +48,7 @@ public User getUserWithCredential(String credential) throws ValidationException
         return getUserWithUsername(username);
     }
 
+    // TODO: should we throw a different exception if the user is not found?
```

Comment:
Good point. We should implement more specific exceptions instead of all validation exceptions.